packages/react-components/src/components/table-output-component.tsx

```diff
@@ -653,4 +655,143 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
+
+                const endTimeStampExport = (this.startTimestamp > this.endTimestamp ? this.startTimestamp: this.endTimestamp) + BigInt(1);
+                if (endTimeStampExport <= this.props.range.getEnd()) {
+                    endIndex = await this.fetchTableIndex(endTimeStampExport);
+                    if (endIndex) {
+                        endIndex = endIndex - 1;
+                    }
+                } else {
+                    endIndex = this.props.nbEvents - 1;
+                }
+
+                if (startIndex !== undefined && endIndex !== undefined) {
+                    fetchIndex = startIndex;
+                    totalLinesToFetch = endIndex - startIndex + 1;
+                }
+            }
+        }
+
+        let fetchLinesRemainder = totalLinesToFetch;
+
+        while (fetchLinesRemainder > 0 && this.mainOutputContainer.current) {
+            let curLinesToFetch = bufferSize;
+
+            if (fetchLinesRemainder > bufferSize) {
+                fetchLinesRemainder-=bufferSize;
+            } else {
+                curLinesToFetch = fetchLinesRemainder;
+                fetchLinesRemainder = 0;
+            }
+
+            const tspClientResponse = await tspClient.fetchTableLines(traceUUID, outputId, QueryHelper.tableQuery(this.columnIds, fetchIndex, curLinesToFetch,
+                { [QueryHelper.REQUESTED_TABLE_COLUMN_IDS_KEY]: columnsIds }));
+            const lineResponse = tspClientResponse.getModel();
+            if (!tspClientResponse.isOk() || !lineResponse) {
+                return;
+            }
+            const model = lineResponse.model;
+            const lines = model.lines;
+            lines.forEach(line => {
+                const obj: any = {};
+                const cells = line.cells;
+                const columnIds = model.columnIds;
+
+                if (this.showIndexColumn) {
+                    obj[0] = line.index.toString();
+                }
+
+                cells.forEach((cell, index) => {
+                    const id = this.showIndexColumn ? columnIds[index] + 1 : columnIds[index];
+                    obj[id] = cell.content;
+                });
+
+                linesArray.push(obj);
+            });
+
+            for (let i=0;i<lines.length;i++) {
+                // Stores each csv row data
+                const csvrow = [];
+                for (let j=0;j<lines[i].cells.length;j++) {
+                    // Get the text data of each cell of
+                    // a row and push it to csvrow
+                    const cellContent = lines[i].cells[j].content;
+                    if (cellContent.includes(',')) {
+                        csvrow.push('"' + cellContent + '"');
+                    } else {
+                        csvrow.push(cellContent);
+                    }
+                }
+                // Combine each column value with comma
+                csv_data.push(csvrow.join(','));
```

Comment:
It's not acceptable to accumulate all rows in memory, it can easily crash the browser with a large trace. The partial data should be appended to the target file at each iteration and discarded from memory.